OgreMain/src/OgreRenderSystem.cpp

```diff
@@ -618,10 +618,9 @@ namespace Ogre {
     void RenderSystem::shutdown(void)
     {
         // Remove occlusion queries
-        for (HardwareOcclusionQueryList::iterator i = mHwOcclusionQueries.begin();
-            i != mHwOcclusionQueries.end(); ++i)
+        for (auto &a : mHwOcclusionQueries)
```

Comment:
```suggestion
        for (auto q : mHwOcclusionQueries)
```
as this a pointer, `auto&` does not save anything. Also `q` for query is more descriptive as a running variable

Reply:
but in this case it makes copy, using reference wouldn't be more efficient ?

Reply:
no, a reference has the same cost as copying a pointer. Sometimes more as a copy tells the compiler that the original cannot change.